File: 5-python/py/packages/addr/set.py
import nimbella
import json

def main(args):
    db = nimbella.redis()
    key = "address:"+args.get("name", "")
    value = json.dumps({
            "name": args.get("name", ""),
            "company":  args.get("company", ""),
            "phone": args.get("phone", "")
    })
    return { "body": db.set(key, value) }

File: 6-typescript/src/set.ts
import nim = require("@nimbella/sdk")

interface Record {
    name: string,
    company: string,
    phone: number
}

function set(args: Map<string, string>) {
    let db = nim.redis()
    let key = "address:"+args["name"]
    let value: Record = {
            "name": args["name"] || "",
            "company":  args["company"]  || "",
            "phone": args["phone"] || ""
    }
    return db.setAsync(key, JSON.stringify(value))
    .then(reply => { return {"body": reply}})
    .catch(err =>  { return {"body": err}})
}
